azure-functions/acbs-function/src/functions/activity-update-facility-guarantee.js

```diff
@@ -0,0 +1,80 @@
+/*
+ * Facility guarantee amendment DAF
+ * ***************************
+ * This DAF (Durable Activity Function) is never invoked directly.
+ * It is invoked via DOF (Durable Orchestrator Function).
+ *
+ * Pre-requisites
+ * --------------
+ * 0. 'npm install durable-functions'
+ * 1. Durable  HTTP trigger function (acbs-http)
+ * 2. Durable Orchestrator function (DOF) (acbs-amend-facility)
+ *
+ * ------------------
+ * HTTP -> DOF -> DAF
+ * ------------------
+ */
+const df = require('durable-functions');
+const { getNowAsIsoString } = require('../../helpers/date');
+const api = require('../../api');
+const { isHttpErrorStatus } = require('../../helpers/http');
+
+/**
+ * This function is used to update the guarantee of a facility. It first checks if the payload is valid and contains all mandatory fields.
+ * If the payload is valid, it sends a request to the API to update the facility guarantee.
+ * If the API request is successful, it returns an object containing the status, timestamps of when the request was sent and received, the data sent, and the data received from the API.
+ * If the API request fails, it throws an error with details about the request and the error.
+ * If the payload is not valid or does not contain all mandatory fields, it returns an object with the missing mandatory fields.
+ * If any other error occurs, it logs the error and throws a new error.
+ *
+ * @param {object} payload - The payload containing the facilityIdentifier and acbsFacilityGuaranteeInput.
+ * @param {string} payload.facilityIdentifier - The identifier of the facility.
+ * @param {object} payload.acbsFacilityGuaranteeInput - The acbsFacilityGuaranteeInput object containing the guarantee details.
+ * @returns {object} - An object containing the status, timestamps of when the request was sent and received, the data sent, and the data received from the API.
+ * @throws {Error} - Throws an error if the payload is invalid, if the API request to generate the guarantee ID fails, or if any other error occurs.
+ */
+const handler = async (payload) => {
+  try {
+    if (!payload) {
+      throw new Error('Invalid facility guarantee update payload');
+    }
+
+    const { facilityIdentifier, acbsFacilityGuaranteeInput } = payload;
+
```

Comment:
Missing mandatory field check, please refer to `activity-update-facility-loan-amount.js`

```javascript
const missingMandatory = findMissingMandatory(acbsFacilityLoanInput, mandatoryFields);
```

Reply:
Three are no mandatory fields here -- both are optional

Reply:
Understood, maybe we should do below to ensure we are not making an API call without any fields to get `400` back. What are your thoughts?
let mandatoryFields = [];

```javascript
if (acbsFacilityGuaranteeInput.expirationDate) {
mandatoryFields.push('expirationDate');
}

if (acbsFacilityGuaranteeInput.amount) {
mandatoryFields.push('amount');
}
```

Reply:
We had a similar converstaion for the previous 6616 ticket, where it was decided we'd add this verification into the `facility-covenant-amend` file, so I've added the similar into the `facility-guarantee-amend` file
```
    if (!amount && !facilityGuaranteeDates?.guaranteeExpiryDate) {
      throw new Error('Invalid argument set provided');
    }
```

The suggested code
```
if (acbsFacilityGuaranteeInput.fieldExample) {  
mandatoryFields.push('fieldExample');  
}  
```
wouldn't do anything I think? As mandatoryField function just checks to see if `acbsFacilityGuaranteeInput` contains the fields passed in. As we are driving `mandatoryFields` by what exists in `acbsFacilityGuaranteeInput`, we are saying "Heres a list of fields that I know exist in `acbsFacilityGuaranteeInput` as that is how I've got the list of them, please check they exist in `acbsFacilityGuaranteeInput`

Reply:
There is an issue here with existing patterns and how we handle errors. As both fields are mandatory, the agreed patterns for where we look to validate means that this will happily fail silently. 

I'm going to update this to work for now and try to stick to the existing error handling patterns as much as possible, but this will greatly limit where I can add these checks. I've added jsdoc to clarify the structure of the mapped amendments at any point.

Reply:
Good shout on updating the JSDocs, agree on the limitation.